Lib/email/utils.py

```diff
@@ -259,21 +259,13 @@ def decode_params(params):
 
     params is a sequence of 2-tuples containing (param name, string value).
     """
-    # Copy params so we don't mess with the original
-    params = params[:]
-    new_params = []
+    new_params = [params[0]]
```

Comment:
What if `params[0]` is not a 2-tuple?

Reply:
That's possible for the first params element, though the others are implicitly checked by the unpacking for-loop.

Looking at the usage of `decode_params`, it's only use is in `email.message._get_params_preserve` which in turn has 3 call sites. All 3 calls unpack the elements as `(k, v)` pairs. The bad data would cause an immediate error, given how the method is used. The error would be the same as today, though the traceback would not be as deep.

If that's not satisfactory, I'll revert to:

    key, value = params[0]
    new_params = [(key, value)]